packages/casadi/test_casadi.py

```diff
@@ -0,0 +1,298 @@
+import pytest
+from pytest_pyodide import run_in_pyodide
+
+
+@run_in_pyodide(packages=["casadi"])
+def test_symbolic_variable(selenium):
+    import casadi as ca
+
+    x = ca.SX.sym("x")
+    assert isinstance(x, ca.SX)
+    assert x.name() == "x"
+
+
+@run_in_pyodide(packages=["casadi", "numpy"])
+def test_basic_casadi_function_eval(selenium):
+    import casadi as ca
+    import numpy as np
+
+    x = ca.MX.sym("x")
+    y = ca.MX.sym("y")
+    f = ca.Function("f", [x, y], [x**2 + y**2])
+
+    assert np.allclose(f(1, 2).full(), np.array([5]))
+    assert np.allclose(f(3, 4).full(), np.array([25]))
+    assert np.allclose(f(5, 6).full(), np.array([61]))
+
+
+# IPOPT not available because compiled with -DWITH_IPOPT=OFF
+# by default. This can be changed by setting -DWITH_IPOPT=ON
+# in the future
```

Comment:
Why haven't you turned on IPOPT? Does compilation fail? Take a very long time? Make a much larger wheel?

Reply:
Nevermind, should have read the PR description. Can we say that we compile without IPOPT for size reasons here?

Reply:
Yes, sure. The primary reason was size, so I didn't actually try to compile with IPOPT enabled – if it works, great, it shouldn't increase the wheel size that much. I just downloaded the manylinux amd64 wheel and the bulk of its size, i.e., the largest shared object files (in descending order) in the contents are all OpenBLAS ones (shrug), followed by those for HiGHS, IPOPT, Alpaqa, MUMPS, Spral, and BONMIN, and then the rest of CasADi's own extension modules (plugins) come up. The WASM wheel doesn't have OpenBLAS, so the wheel size without all these extra solvers is just ~2.1 MB.

Reply:
Also, the default 32-bit build flags disable some of the other features besides these, like BLASFEO: https://github.com/casadi/casadi/blob/3d820e62cb588e76498c92bf6feff6876b4e7ca0/.github/workflows/binaries.yml#L8

Reply:
Truncated output after I tried to build with IPOPT enabled:

<details>
<summary>Tap to view logs</summary>

```
running bdist_wheel                                                                                                                                      
running build                                                                                                                                            
running build_ext                                                                                                                                        
configure: cmake -DCMAKE_C_COMPILER=/tmp/tmp58z8w9qn/cc -DCMAKE_CXX_COMPILER=/tmp/tmp58z8w9qn/c++ -DCMAKE_AR=/tmp/tmp58z8w9qn/ar                         
-DCMAKE_C_COMPILER_AR=/tmp/tmp58z8w9qn/ar -DCMAKE_CXX_COMPILER_AR=/tmp/tmp58z8w9qn/ar --fresh --version                                                  
-DCMAKE_CROSSCOMPILING_EMULATOR=/src/emsdk/emsdk/node/18.20.3_64bit/bin/node                                                                             
<string>:30: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.                                                
configure: cmake -DCMAKE_C_COMPILER=/tmp/tmp58z8w9qn/cc -DCMAKE_CXX_COMPILER=/tmp/tmp58z8w9qn/c++ -DCMAKE_AR=/tmp/tmp58z8w9qn/ar                         
-DCMAKE_C_COMPILER_AR=/tmp/tmp58z8w9qn/ar -DCMAKE_CXX_COMPILER_AR=/tmp/tmp58z8w9qn/ar --fresh /src/packages/casadi/build/casadi-3.6.5 -DSWIG_IMPORT=ON   
-DWITH_PYTHON=ON -DWITH_EXAMPLES=OFF -DPYTHON_INCLUDE_DIR=/src/cpython/installs/python-3.12.1/include/python3.12                                         
-DCMAKE_INSTALL_PREFIX=/src/packages/casadi/build/casadi-3.6.5/build/lib.emscripten_3_1_58_wasm32-cpython-312 -DWITH_SELFCONTAINED=ON -DWITH_DEEPBIND=ON 
-DWITH_PYTHON3=ON -DCMAKE_CROSSCOMPILING_EMULATOR=/src/emsdk/emsdk/node/18.20.3_64bit/bin/node                                                           
-- No flag needed for enabling C++11 features.                                                                                                           
EXTERNAL_CMAKE_COMMAND: /usr/local/lib/python3.12/site-packages/cmake/data/bin/cmake                                                                     
CXX /tmp/tmp58z8w9qn/c++                                                                                                                                 
FC /tmp/tmp58z8w9qn/gfortran                                                                                                                             
CC /tmp/tmp58z8w9qn/cc                                                                                                                                   
-- Could NOT find Ipopt (missing: Ipopt_DIR)                                                                                                             
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.1")                                                                                          
-- Checking for one of the modules 'ipopt'                                                                                                               
-- Could NOT find IPOPT (missing: IPOPT_LIBRARIES IPOPT_INCLUDE_DIRS)                                                                                    
CMake Error at CMakeLists.txt:933 (message):                                                                                                             
  IPOPT not found on your system yet required.  Set WITH_BUILD_IPOPT or                                                                                  
  WITH_BUILD_REQUIRED ON to build this required package from source.                                                                                     
                                                                                                                                                         
                                                                                                                                                         
-- Configuring incomplete, errors occurred!                                                                                                              
emcmake: error: 'cmake -DCMAKE_C_COMPILER=/tmp/tmp58z8w9qn/cc -DCMAKE_CXX_COMPILER=/tmp/tmp58z8w9qn/c++ -DCMAKE_AR=/tmp/tmp58z8w9qn/ar                   
-DCMAKE_C_COMPILER_AR=/tmp/tmp58z8w9qn/ar -DCMAKE_CXX_COMPILER_AR=/tmp/tmp58z8w9qn/ar --fresh /src/packages/casadi/build/casadi-3.6.5 -DSWIG_IMPORT=ON   
-DWITH_PYTHON=ON -DWITH_EXAMPLES=OFF -DPYTHON_INCLUDE_DIR=/src/cpython/installs/python-3.12.1/include/python3.12                                         
-DCMAKE_INSTALL_PREFIX=/src/packages/casadi/build/casadi-3.6.5/build/lib.emscripten_3_1_58_wasm32-cpython-312 -DWITH_SELFCONTAINED=ON -DWITH_DEEPBIND=ON 
-DWITH_PYTHON3=ON -DCMAKE_CROSSCOMPILING_EMULATOR=/src/emsdk/emsdk/node/18.20.3_64bit/bin/node' failed (returned 1)
```

</details>

I guess we need to build and provide IPOPT separately, just like [NLOPT](https://github.com/pyodide/pyodide/tree/main/packages/nlopt) has been done. I would like to suggest keeping this for another PR. Should we keep the test, though, or should I comment it out? I assume I'll encounter similar errors with the rest of the solvers/interfaces.

Reply:
I don't really care whether you keep the test or not. It might be slightly better to delete tests rather than xfailing them but the xfailed test does document that the functionality is missing.

Reply:
I think I'll comment it out in that case, so that no one is misled by the skip message when they browse the collective logs

Reply:
Done in d8b140180f8acefb5fd4c48fc543fee035e7908b.